### PR TITLE
MacOS AWT/Swing support, JDK-7131356 workaround

### PR DESCRIPTION
**Pull request by dbarganski**

### CODE REVIEW

**Comment by timob** [comment]
Nice 👍! I will try this out so I can understand how this works. Is it possible to post some example code that uses AWT/Swing with JNIGI?

File: README.md [hunk 1]
```diff
@@ -25,13 +25,28 @@ The JVM calls sigaction, Golang requires that SA_ONSTACK flag be passed.
...
 
+### MacOS AWT/Swing
+
+In order to run AWT/Swing on MacOS (only) is requires to run a platform specific Cocoa Run Loop code otherwise java gui becomes irresponsive.
+
+You would need to:
+
+1) Spin go routine in a *separate* thread from which is initialized JVM 
+2) In the application's main thread run `RunJVMGUILoop` function which configures and executes Cocoa run loop. 
+Running the loop is a blocking opperation for the current (main) thread.
```
Comment (timob) [inline README.md:36]: If I am understanding this correctly the RunJVMLoop is the UI thread which handles events, like button clicks etc. So any non-UI code your App runs is in separate thread. Is that right?

This is similar to many UI toolkits, you can see how the UI loop works in SWT:
https://github.com/timob/swt/blob/master/example/example1.go#L97

**Comment by timob** [comment]
About https://github.com/timob/jnigi/issues/35, that you are fixing in this PR too. I just tried using `go test` and also ran this graphical program https://github.com/timob/swt/blob/master/example/example1.go .

I'm not getting the popup window or the error message. Maybe this issue has been fixed in OpenJDK.

What versions are you using? I'm using `go version go1.13.5 darwin/amd64` with 
```
openjdk version "14.0.2" 2020-07-14
OpenJDK Runtime Environment (build 14.0.2+12-46)
OpenJDK 64-Bit Server VM (build 14.0.2+12-46, mixed mode, sharing)
```
on Mojave

**Comment by dbarganski** [comment]
Probably you have an installed JDK? In my use case i have JDK contained within project's sub-directory.

openjdk 11.0.1 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)

File: README.md (hunk 1 above)
Comment (dbarganski) [reply README.md:36]: Separate thread and runloop is only requirement on mac and only if you use awt. Also some GUI libraries (do not have reference now) must not use this approach. this is reason for java launcher -XstartOnFirstThread [link](https://github.com/AdoptOpenJDK/openjdk-jdk/blob/704f784c88ee282837c980948167e741e7227f27/src/java.base/macosx/native/libjli/java_md_macosx.m#L866)

* Apple has excellent document describing the requirement and iterations:

[Section "Invoking the Java Virtual Machine From Native Code"](https://developer.apple.com/library/archive/technotes/tn2147/_index.html#//apple_ref/doc/uid/DTS10003827-CH1-INVOCATION)

* My code is based on [official Apple sample](https://developer.apple.com/library/archive/samplecode/simpleJavaLauncher/Introduction/Intro.html#//apple_ref/doc/uid/DTS10000688)

* OpenJDK java launcher is also using a very similar approach (https://github.com/AdoptOpenJDK/openjdk-jdk/blob/master/src/java.base/macosx/native/libjli/java_md_macosx.m#L354) 

* Good reference on the topic 

[OSX: JavaVM, AWT/Swing and possibly a deadlock](https://stackoverflow.com/questions/8750690/osx-javavm-awt-swing-and-possibly-a-deadlock/14859179)

A bit twisted setup on mac :)

Hope it helps as justification. My main purpose is to *allow* using AWT on mac by providing necessary code but it should be an optional.

**Comment by timob** [comment]
I could have a system  JDK installed, when I installed Intelli-J IDE it installed OpenJDK in `~/Library/Java/JavaVirtualMachines/openjdk-14.0.1` and this seems to be available system wide.

I'm not sure how to uninstall it. So let's just do the fix, and see how it works.

I've split the bug fix out into this PR https://github.com/timob/jnigi/pull/43, can you 👀 at that PR?

**Comment by timob** [comment]
JDK-7131356 fix is now in master.

**Comment by timob** [comment]
Most of this is merged. Please start new PR for remaining part if needed.

**Pull request closed without merging**